rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/kustomize/KustomizeTemplateUtils.java

```diff
@@ -79,11 +99,71 @@ public BakeRecipe buildBakeRecipe(
     command.add("kustomize");
     command.add("build");
     command.add(templatePath.getParent().toString());
+
+    BakeRecipe result = new BakeRecipe();
     result.setCommand(command);
+    return result;
+  }
+
+  private BakeRecipe buildBakeRecipeFromGitRepo(
+      BakeManifestEnvironment env, KustomizeBakeManifestRequest request, Artifact artifact)
+      throws IOException {
+    // This is a redundant check for now, but it's here for when we soon remove the old logic of
+    // building from
+    // a github/file artifact type and instead, only support the git/repo artifact type
+    if (!"git/repo".equals(artifact.getType())) {
+      throw new IllegalArgumentException("The inputArtifact should be of type \"git/repo\".");
+    }
+
+    String kustomizeFilePath = request.getKustomizeFilePath();
+    if (kustomizeFilePath == null) {
+      throw new IllegalArgumentException("The bake request should contain a kustomize file path.");
+    }
+
+    Path tarPath = env.resolvePath("repo.tar.gz");
+    try {
+      artifactDownloader.downloadArtifact(artifact, tarPath);
+    } catch (IOException e) {
+      throw new IOException("Failed to download git/repo artifact: " + e.getMessage(), e);
+    }
 
+    try {
+      extractArtifact(tarPath, env.resolvePath(""));
+    } catch (IOException e) {
+      throw new IOException("Failed to extract git/repo artifact: " + e.getMessage(), e);
+    }
+
+    List<String> command = new ArrayList<>();
+    command.add("kustomize");
+    command.add("build");
+    command.add(env.resolvePath(kustomizeFilePath).getParent().toString());
+
+    BakeRecipe result = new BakeRecipe();
+    result.setCommand(command);
     return result;
   }
 
+  // This being here is temporary until we find a better way to abstract it
+  public static void extractArtifact(Path inputTar, Path outputPath) throws IOException {
+    TarArchiveInputStream tarArchiveInputStream =
+        new TarArchiveInputStream(
+            new GzipCompressorInputStream(new BufferedInputStream(Files.newInputStream(inputTar))));
+
+    ArchiveEntry archiveEntry = null;
+    while ((archiveEntry = tarArchiveInputStream.getNextEntry()) != null) {
+      Path archiveEntryOutput = outputPath.resolve(archiveEntry.getName());
```

Comment:
How does this code behave if the archive entry's name references the parent directory?  Could this break out of the temporary directory we're trying to extract to?

Reply:
I tested this out and you're right that if I force an entry to reference a parent, it does break out. Is this something that we want to support and handle in some way? Or should it be considered an error if an entry at the repo level tries to reference a parent? Because in theory, we wouldn't even have access to those files, correct?

Reply:
I think we'd want to immediately throw an exception and abort if the entry tries to reference something outside of the repo directory.  This would avoid security vulnerabilities where someone could send a specially-crafted repo artifact that unpacks on top of system files.  (You're right that in theory important files shouldn't be accessible by the `rosco` process unless it's running as `root` which it shouldn't, but I still think it's better to fail fast before depending on these permissions being set correctly.)

A good example where we do this check elsewhere and fail fast is [here](https://github.com/spinnaker/rosco/blob/14b794b7b09ab2516bbb7169457c9dd2a8dda4bc/rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/BakeManifestEnvironment.java#L53).  (That example is not in a re-usable place, but you could do something similar in the function you wrote.)

Reply:
Yep, that makes a lot of sense. 👍